core/drivers/crypto/caam/utils/utils_delay.c

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018 NXP
+ *
+ * Brief   Delay management utilities.
+ *         Primitive to delay a delay.
+ */
+#include <arm.h>
+#include <caam_utils_delay.h>
+#include <types_ext.h>
+
+/*
+ * Wait given microsecond
+ *
+ * @time  Time in microsecond
+ */
+void caam_udelay(uint32_t time)
```

Comment:
How about using `udelay()` instead?

Reply:
`udelay()` currently uses the core internal timer, which is not available on Cortex-A9 cores, i.e. i.MX6Q.

Reply:
Aha, missed that. Perhaps the platform should be able to override `udelay()`. Because this implementation doesn't seem very suitable for generic code.

Reply:
As this is specific to NXP CAAM driver I prefer to not change it and it will be reviewed later when we will finish the NXP i.MX implementation support for all device type (MX6/7/8).

Reply:
OK